examples/plot_barber2020_simulations.py

```diff
@@ -103,61 +123,75 @@ def plot_simulation_results(results: Dict, methods: List[str], title: str) -> No
         Title of the plot.
     """
     fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(15, 5))
-    plt.rcParams.update({'font.size': 14})
+    plt.rcParams.update({"font.size": 14})
     if title is not None:
         plt.suptitle(title)
     for method in methods:
-        coverage_mean = []
-        coverage_SE = []
-        for dim_val in dim_vals:
+        coverage_mean, coverage_SE = [], []
+        width_mean, width_SE = [], []
+        for dimension in dimensions:
             coverage_mean.append(np.array([
-                results[key][3] for key in results
-                if (results[key][2] == method) & (results[key][1] == dim_val)
+                results[key]["coverage"] for key in results
+                if (
```

Comment:
C'est vraiment très bizarre ce `if`. Peux-tu m'expliquer pourquoi il est nécessaire ? Je pense que l'on peut adapter la structure de `results` pour itérer dessus sans cette complexité. Peut-être qu'il faut aborder le sujet de vive voix pour être efficace.

Reply:
Avec un `DataFrame` ça se fait en une ligne de code mais en utilisant uniquement des `Dict` ça devient en effet plus compliqué...

Reply:
`results` est désormais un nested dictionary afin de simplifier le calcul des coverages et des widths pour la figure.